### PR TITLE
Some doc fixes for Flask-Restful

### PR DESCRIPTION
**Pull request by ueg1990**

### CODE REVIEW

File: docs/extending.rst [hunk 1]
```diff
@@ -116,14 +116,13 @@ These output functions take three parameters, ``data``, ``code``, and
 ``headers``
 
 ``data`` is the object you return from your resource method, code is the HTTP
-status code that it expects, and headers are any HTTP headers to set in the
+status code that it expects, and headers are any HTTP headers to define in the
```
Comment (joshfriend) [inline docs/extending.rst]: "set" is the typical term used for this.

File: docs/fields.rst [hunk 2]
```diff
@@ -8,7 +8,7 @@ Output Fields
 
 Flask-RESTful provides an easy way to control what data you actually render in
 your response.  With the ``fields`` module, you can use whatever objects (ORM
-models/custom classes/etc) you want in your resource and ``fields`` lets you
+models/custom classes/etc.) you want in your resource; ``fields`` also lets you
```
Comment (joshfriend) [inline docs/fields.rst]: needs a period instead of semicolon.

File: docs/extending.rst (hunk 1 above)
Comment (ueg1990) [inline docs/extending.rst]: ok, i will revert back that change...let me know if there are any other changes before i update the pull request. thanks for the quick review

File: docs/fields.rst [hunk 3]
```diff
@@ -48,15 +48,14 @@ The decorator ``marshal_with`` is what actually takes your data object and appli
 field filtering.  The marshalling can work on single objects, dicts, or
 lists of objects.
 
-Note: marshal_with is a convenience decorator, that is functionally equivalent to the following ``return marshal(db_get_todo(), resource_fields), 200``.
-This explicit expression can be used to return other HTTP status codes than 200 along with a successful response (see ``abort`` for errors).
+Note: marshal_with is a convenience decorator, that is functionally equivalent to ``return marshal(db_get_todo(), resource_fields), 200``.
```
Comment (joshfriend) [inline docs/fields.rst]: The code example here is probably best formatted as a separate code block instead of inline with the paragraph.

Comment (ueg1990) [inline docs/fields.rst]: i had thought of doing this but was not sure....will do it now.

**Comment by joshfriend** [comment]
Please reformat some of your changes to wrap at the 80 character column so the formatting of the document source stays consistent with what's there already. You can typically configure your editor to show this. For example if you use Sublime Text, select "View" -> "Ruler" -> "80" from the menu.

File: docs/extending.rst (hunk 1 above)
Comment (joshfriend) [inline docs/extending.rst]: I'm still looking through it. I'll leave a comment on the PR when I'm done.

File: docs/fields.rst [hunk 4]
```diff
@@ -144,7 +143,7 @@ the default scheme, pass the ``scheme`` keyword argument::
 Complex Structures
 ------------------
 
-You can have a flat structure that marshal_with will transform to a nested structure ::
+You can have a flat structure that the function ``marshal`` will transform to a nested structure ::
```
Comment (joshfriend) [inline docs/fields.rst]: Try:

```rst
You can have a flat structure that :meth:`~flask.ext.restful.marshal` will transform to a nested structure
```

Includes a link to the function that way.

File: docs/fields.rst [hunk 5]
```diff
@@ -210,10 +209,10 @@ structures and render them appropriately. ::
     '{"billing_address": {"line 1": "123 fake street", "line 2": null, "state": "NY", "zip": "10468", "city": "New York"}, "name": "bob", "shipping_address": {"line 1": "555 nowhere", "line 2": null, "state": "NY", "zip": "10468", "city": "New York"}}'
 
 This example uses two Nested fields.  The ``Nested`` constructor takes a dict of
-fields to render as sub-fields.  The important difference when using ``Nested``
-vs nested dicts as the previous example is the context for attributes.  In
-this example "billing_address" is a complex object that has it's own fields,
-the contexted passed to the nested field is the sub-object instead of the
+fields to render as sub-fields.  The important difference between the ``Nested`` constructor
```
Comment (joshfriend) [inline docs/fields.rst]: Can you make the first reference to `Nested` in this section a reference to the class docs?

File: docs/fields.rst [hunk 6]
```diff
@@ -210,10 +209,10 @@ structures and render them appropriately. ::
...
+and nested dicts (previous example), is the context for attributes.  In
+this example, "billing_address" is a complex object that has its own fields;
```
Comment (joshfriend) [inline docs/fields.rst]: replace semicolon with "and"?

File: docs/intermediate-usage.rst [hunk 7]
```diff
@@ -132,12 +132,11 @@ exercise a larger amount of options. We'll define a resource named "User". ::
...
         @marshal_with(user_fields)
         def get(self, id):
-            args = get_parser.parse_args()
+            args = post_parser.parse_args()
             user = fetch_user(id)
             return user
 
-As you can see, we create a `post_parser` specifically to handle parsing
-the arguments provided on POST. Let's step through the definition of each
+As you can see, we create a `post_parser` specifically to handle the parsing of arguments provided on POST. Let's step through the definition of each
```
Comment (joshfriend) [inline docs/intermediate-usage.rst]: Another thing you can fix is that variable names in paragraphs need double backticks to be formatted as `monospaced`

File: docs/intermediate-usage.rst [hunk 8]
```diff
@@ -213,9 +212,9 @@ way that it would be an acceptable argument to `marshal` by itself.
...
 Finally, we used the `fields.Url` field type. ::
 
-        'friends': fields.Url('/Users/{id}/Friends', absolute=True),
-        'posts': fields.Url('Users/{id}/Posts', absolute=True),
+        'friends': fields.Url('/users/{id}/Friends', absolute=True),
+        'posts': fields.Url('/users/{id}/Posts', absolute=True),
 
 It takes a string that can be formatted in the same manner as `fields.FormattedString`
-which we covered above.  Passing `absolute=True` ensures that the generated Urls
+which we covered above.  Passing `absolute=True` ensures that the generated urls
```
Comment (joshfriend) [inline docs/intermediate-usage.rst]: double backticks around `absolute=True`...

**Comment by joshfriend** [comment]
OK I think that's everything I can think of. Wow, there were lots of stupid grammar issues in the original! Thanks so much for taking time to help fix this up!

**Comment by ueg1990** [comment]
my please...can you help me with one thing: how can i add reference to Nested to the class docs? should i do: :py:class:`Nested`? 
also how can i view the changes for .rst files in the format as the flask-restful doc....i can open them in the browser?

**Comment by joshfriend** [comment]
Use:
```rst
:class:`~fields.Nested`
```

You can build the docs with the command:

```
make doc
```

or alternatively:

```
make read
```

 which will build the docs, then open them in your browser. `virtualenv` is required for this. I haven't tested it recently on windows, but it should work.

**Comment by ueg1990** [comment]
it worked like a charm :+1:

**Comment by joshfriend** [comment]
Thanks again! :+1:

**Comment by ueg1990** [comment]
hey josh, i had a quick question. if i want to test changes in the code for flask-restful....how can i test them? do i have to do something like: "python setup.py develop" or "make develop" and then import it in a test file?

**Comment by joshfriend** [comment]
[Check out the testing page in the docs](http://flask-restful.readthedocs.org/en/latest/testing.html)

**Pull request merged**